internal/test/wsu/wsu_test.go

```diff
@@ -164,53 +164,73 @@ func testBuiltWMCB(t *testing.T, ansibleOutput string) {
 func TestWSU(t *testing.T) {
 	require.NotEmptyf(t, playbookPath, "WSU_PATH environment variable not set")
 	require.NotEmptyf(t, clusterAddress, "CLUSTER_ADDR environment variable not set")
-
 	require.GreaterOrEqual(t, vmCount, 1, "Expected one or more VMs")
 
 	wmcbPinnedURL, err := getLatestReleasedArtifactURL("wmcb.exe")
 	require.NoError(t, err, "Could not get WMCB url")
-	pinnedWMCBOption := "wmcb_url=" + wmcbPinnedURL
 
 	// Run VM specific tests
 	for i, vm := range framework.WinVMs {
-		func(vmNum int, vm e2ef.WindowsVM) {
-			t.Run("VM "+strconv.Itoa(i), func(t *testing.T) {
-				// Indicate that we can run the test suite on each node in parallel
-				t.Parallel()
-
-				// In order to run the ansible playbook we create an inventory file:
-				// https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
-				// Passing in a slice with a single entry for now
-				hostFilePath, err := createHostFile([]e2ef.WindowsVM{framework.WinVMs[vmNum]})
-				require.NoErrorf(t, err, "Could not write to host file: %s", err)
-
-				// Run the playbook
-				// For the first VM we use a pinned WMCB version
-				var ansibleCmd *exec.Cmd
-				var pinnedWMCB bool
-				if vmNum == 0 {
-					pinnedWMCB = true
-					ansibleCmd = exec.Command("ansible-playbook", "-v", "-e", pinnedWMCBOption, "-i",
-						hostFilePath, playbookPath)
-				} else {
-					ansibleCmd = exec.Command("ansible-playbook", "-v", "-i", hostFilePath, playbookPath)
-				}
-				out, err := ansibleCmd.CombinedOutput()
-				require.NoError(t, err, "WSU playbook returned error: %s, with output: %s", err, string(out))
-				runE2ETestSuite(t, pinnedWMCB, vm, string(out))
-
-				// Run the test suite twice, to ensure that the WSU can be run multiple times against the same VM
-				t.Run("Run the WSU against the same VM again", func(t *testing.T) {
-					runE2ETestSuite(t, pinnedWMCB, vm, string(out))
-				})
-			})
-		}(i, vm)
+		t.Run("VM "+strconv.Itoa(i), func(t *testing.T) {
+			testVM(t, i, vm, wmcbPinnedURL)
+		})
 	}
 
 	// Run cluster wide tests
 	t.Run("Pending CSRs were approved", testNoPendingCSRs)
 }
 
+// testVM runs the WSU against the given VM and runs the e2e test suite against that VM as well
+func testVM(t *testing.T, vmNum int, vm e2ef.WindowsVM, wmcbURL string) {
+	var wsuOut string
+	var pinnedWMCB bool
+	var err error
+	pinnedWMCBOption := "wmcb_url=" + wmcbURL
+
+	// Indicate that we can run the test suite on each node in parallel
+	t.Parallel()
+
+	// Run the WSU against the VM
+	if vmNum == 0 {
+		// For the first VM we use a pinned WMCB version
+		wsuOut, err = runWSUAgainstVM(vm, pinnedWMCBOption)
```

Comment:
The way, I had in my head was to run WSU once against both the VMs. I think, that will test the scenario where we've 2 host entries in the inventory file.